cmake-modules/AzureCodeCoverage.cmake

```diff
@@ -55,7 +55,8 @@ function(add_gcovr_html)
             -r ${BASEDIR} ${EXCLUDE_ARGS}
             --object-directory=${PROJECT_BINARY_DIR}
             -o ${args_TARGET_NAME}/index.html
-
+            --exclude-unreachable-branches
```

Comment:
These would be dead code? Would be great if we could remove em

Reply:
These are *not* dead code. These are code branches generated by the compiler which can never be executed.

There's more information in the FAQ about them (https://gcovr.com/en/stable/faq.html#why-does-c-code-have-so-many-uncovered-branches).

Reply:
Maybe it's worth adding the link to this doc as a comment in this source file, for later reference.

Reply:
A good idea, done.